test/unit/chains.test.ts

```diff
@@ -27,6 +27,45 @@ describe('Chain metadata', () => {
         expect(metadata.gasCurrencyCoinGeckoId).not.to.be.undefined;
       }
     });
+
+    // Values derived from reorg period assessment framework
+    // https://www.notion.so/hyperlanexyz/Reorg-period-assessment-framework-1126d35200d680cbb5f2c67b8b492d62
+    describe('Reorg period', () => {
+      // Ensure all Abacus Works mainnets have technicalStack defined
+      it(`${chain} metadata has technicalStack defined if deployer is Abacus Works, and it is a mainnet with a mailbox deployed`, () => {
+        if (metadata.deployer?.name === "Abacus Works" && !metadata.isTestnet && chainAddresses[chain]?.mailboxAddress) {
+          expect(metadata.technicalStack).not.to.be.undefined;
+        }
+      });
+
+      // Only enforce mainnet reorg period values
+      if (!metadata.isTestnet) {
+        it(`${chain} metadata has reorgPeriod of 5 if technicalStack is opstack`, () => {
+          if (metadata.technicalStack === ChainTechnicalStack.OpStack) {
+            expect(metadata.blocks?.reorgPeriod).to.equal(5);
+          }
+        });
+
+        it(`${chain} metadata has reorgPeriod of 5 if technicalStack is polygoncdk`, () => {
+          if (metadata.technicalStack === ChainTechnicalStack.PolygonCDK) {
+            expect(metadata.blocks?.reorgPeriod).to.equal(5);
+          }
+        });
+
+        it(`${chain} metadata has reorgPeriod of 5 if technicalStack is polkadotsubtrate`, () => {
+          if (metadata.technicalStack === ChainTechnicalStack.PolkadotSubstrate) {
+            expect(metadata.blocks?.reorgPeriod).to.equal(10);
+          }
+        });
+
+        it(`${chain} metadata has reorgPeriod between 0 and 5 if technicalStack is arbitrumnitro`, () => {
+          if (metadata.technicalStack === ChainTechnicalStack.ArbitrumNitro) {
+            expect(metadata.blocks?.reorgPeriod).to.be.at.least(0);
+            expect(metadata.blocks?.reorgPeriod).to.be.at.most(5);
+          }
+        });
```

Comment:
I assume we don't have a defined value for zksync chains yet

Reply:
yeah that's right, I linked the reorg assessment in the description so maybe that's something for myself/trevor/mo to pickup next week